docs/glossary.md

```diff
@@ -0,0 +1,361 @@
+# Glossary
+
+---
+
+## A
+---
+
+#### Account
+
+Accounts allow you to keep separate records of your DCR funds. Transferring DCR across accounts will create a transaction on the blockchain.
+
+#### Address
+
+A secure identifier that represents a possible destination for a payment, similar to an email address. The address is a human-readable representation of the public key receiving payment.
```

Comment:
Addresses are human-readable forms of standard output script types, the most common one being Pay-to-Pubkey-Hash (P2PKH).  In this address format, the address is created by the hash of the public key, not the public key itself, and the redeemer must be able to provide a public key that hashes to a matching hash.  Other types of address formats include Pay-to-Pubkey (P2PK) which works as described here currently, but their usage is not very common, as well as Pay-to-Script-Hash (P2SH) in which more complex and non-standard redeeming requirements can be provided by the redeemer as long as they provide a nested script which hashes to the expected value.

Reply:
This is really good info for a general definition of addresses. But it's not clear to me from this comment which specific method Decred addresses use. Do we use P2PKH? P2PK? iirc, this definition was iterated on in a previous review, and we came up with this, which is accurate but high-level and concise. I think it would be valuable to say technically how Decred generates its hash. A full paragraph about addresses in general is useful as well to the general crypto community, if we don't mind this definition being a bit long. When researching definitions for this glossary, I found that the definitions found in the top glossaries on google were quite lacking, in detail and accuracy.

Reply:
All of them are supported and used.  The most common is P2PKH (starts with Ds), but most VSPs use P2SH (starts with Dc) and the configuration of them uses P2PK (starts with Dk).

Furthermore, there is support for (but nothing really uses it yet) Schnorr signature variants that are different still.

That said, for a glossary, I'm not sure it should go into the specific technical implementation details that are subject to change over time.  Rather, I personally think it should stick to something that more accurately describes their intended purpose as opposed to how they specifically carry it out.  The existing text, I think is pretty close with the exception of the not exactly accurate bits about the public key.

Perhaps:

A human-readable representation of a possible destination for a payment, similar to an email address, with the exception that addresses are typically only intended for a single use.  There are variety of address types that may be used to represent a wide range of constraints required to redeem funds such as requiring a signature from a single party, signatures from multiple parties, or a specific minimum time in the future.

Reply:
After some schooling by @jrick in #documentation yesterday, I feel I mostly understand he above. However, so far the glossary definitions are fairly high-level and user-centric, which is IMO appropriate for dcrdocs. Going too much into the weeds here will confuse users that just want to know how an address works in Decredition, etc. I'm happy to debate, but I would vote for publishing this definition as-is for now and moving debate to this Issue @jholdstock has created.
https://github.com/decred/dcrdocs/issues/710

I do think this is valuable information for a developer. Have copied the above comments to my notes for use in more developer-facing docs.